src/js/tabs/firmware_flasher.js

```diff
@@ -432,27 +408,84 @@ TABS.firmware_flasher.initialize = function (callback) {
+                        if (!supportsUnifiedTargets(descriptor.version)) {
+                            return;
+                        }
+
+                        version = `${version}+${buildNumber}${manufacturerId}`;
+                        build.manufacturerId = manufacturerId;
+                        build.duplicateName = duplicateName;
+                    } else {
+                        version = `${version}+${buildNumber}-legacy`;
+                        build.isLegacy = true;
+                    }
+                    builds[version] = build;
                 });
-                // Assume flashing latest, so default to it.
+            }
+
+            if (callback) {
+                callback();
+            }
+        }
+
+        function populateVersions(versions_element, builds, target) {
+            const sortVersions = function (a, b) {
+                return -semver.compareBuild(a, b);
+            };
+
+            versions_element.empty();
+            const targetVersions = Object.keys(builds);
+            if (targetVersions.length > 0) {
+                versions_element.append($("<option value='0'>{0} {1}</option>".format(i18n.getMessage('firmwareFlasherOptionLabelSelectFirmwareVersionFor'), target)));
+                targetVersions
+                    .sort(sortVersions)
+                    .forEach(function(versionName) {
+                        const version = builds[versionName];
+                        if (!version.isLegacy && !supportsUnifiedTargets(version.descriptor.version)) {
+                            return;
+                        }
+
+                        let versionLabel;
+                        if (version.isLegacy && Object.values(builds).some(function (build) {
+                                return build.descriptor.version === version.descriptor.version && !build.isLegacy;
+                            })) {
+                            versionLabel = i18n.getMessage("firmwareFlasherLegacyLabel", { target: version.descriptor.version });
+                        } else if (!version.isLegacy && Object.values(builds).some(function (build) {
+                                return build.descriptor.version === version.descriptor.version && build.manufacturerId !== version.manufacturerId && !build.isLegacy;
+                            })) {
+                            versionLabel = `${version.descriptor.version} (${version.manufacturerId})`;
+                        } else {
+                            versionLabel = version.descriptor.version;
+                        }
+
+
+                        var select_e =
+                            $("<option value='{0}'>{2} - {1}</option>".format(
+                                    versionName,
+                                    version.descriptor.date,
+                                    versionLabel
+                            ))
+                            .css("font-weight", FirmwareCache.has(version.descriptor)
```

Comment:
Again not your code, but I prefer to add a class and not a style, to maintain it in the css files.